synapse/lib/agenda.py

```diff
@@ -818,12 +780,10 @@ async def _execute(self, appt):
             return
 
         info = {'iden': appt.iden, 'query': appt.query, 'view': view.iden}
-        task = await self.core.boss.execute(self._runJob(user, appt), f'Cron {appt.iden}', user, info=info)
-
-        appt.task = task
-        self._running_tasks.append(task)
 
-        task.onfini(functools.partial(self._running_tasks.remove, task))
+        coro = self._runJob(user, appt)
+        task = self.core.runActiveTask(coro)
+        appt.task = await self.core.boss.promotetask(task, f'Cron {appt.iden}', user, info=info)
```

Comment:
Can we add an ondone handler that clears `appt.task` so that a task can be garbage collected upon completion? Otherwise appt.task will hold a reference to the task until it is run again.

Reply:
We actually may not need that appt.task reference at all if we're holding a reference to the task via `runActiveTask()`